### ISSUE TITLE
build end-to-end timeout for verifiers

### ISSUE THREAD

**Issue by bchurchill** [StanfordPL/stoke#845]
Verification can take unbounded time, even with a timeout on the Solver.  We should really have an end-to-end timeout that kills the verification after a certain point (of course, this will rely on being able to set a timeout on the solver, but what that timeout is depends on how much has elapsed).

**Comment by stefanheule** [comment]
Is this because the solver is invoked many times?

What I've been doing is to use the unix utility "timeout" whenever I invoke stoke_debug_verify, which is probably much more reliable than anything we come up with.  On the other hand it won't work if verification is done as part of search.

**Comment by bchurchill** [comment]
Yes, on both accounts.  Indeed, I'm running the bounded validator in the search loop which is why I could use this.  Although technically, if you put a timeout on the solver the total verification time is still _bounded_, but the bound can be pretty high and hard to reason about -- especially with the alias relationship mining.

**Comment by bchurchill** [comment]
Also, the solver is only invoked multiple times if you have memory or control flow.

**Comment by bchurchill** [comment]
I ended up just using --bound 1 to make the validator finish faster, and that seems to be fixing my real problem.  So, I'm going to downgrade this to wishlist.
